core/templates/pages/exploration-editor-page/services/router.service.ts

```diff
@@ -169,29 +166,23 @@ angular.module('oppia').factory('RouterService', [
     };
 
     var _savePendingChanges = function() {
-      try {
+      // Sometimes, AngularJS throws a "Cannot read property $$nextSibling of
+      // null" error. To get around this we must use $apply().
+      $rootScope.$apply(function() {
         $rootScope.$broadcast('externalSave');
-      } catch (e) {
-        // Sometimes, AngularJS throws a "Cannot read property $$nextSibling of
-        // null" error. To get around this we must use $apply().
-        $rootScope.$apply(function() {
-          $rootScope.$broadcast('externalSave');
-        });
-      }
+      });
     };
 
     var _getCurrentStateFromLocationPath = function() {
-      if ($location.path().indexOf('/gui/') !== -1) {
-        return $location.path().substring('/gui/'.length);
+      var location = $location.path();
+      if (location.indexOf('/gui/') !== -1) {
+        return location.substring('/gui/'.length);
       } else {
         return null;
       }
     };
 
     var _actuallyNavigate = function(pathType, newStateName) {
-      if (pathType !== SLUG_GUI && pathType !== SLUG_PREVIEW) {
```

Comment:
Why delete this?

Reply:
Same doubt.

Reply:
Because `pathType` will always be a SLUG_GUI or a SLUG_PREVIEW. In any _actuallyNavigate calling, `pathType` is either one of these values.